src/linker/Linker/TypeMapInfo.cs

```diff
@@ -299,9 +301,20 @@ IEnumerable<InterfaceImplementation> GetDefaultInterfaceImplementations (TypeDef
 
 		MethodDefinition TryMatchMethod (TypeReference type, MethodReference method)
 		{
+			var isStaticInterfaceMethod = context.TryResolve (method)?.IsStatic;
+			var typeDefintion = context.TryResolve (type);
+			if (isStaticInterfaceMethod == true && !context.Annotations.IsFromStaticInterface (typeDefintion)) {
+				context.Annotations.ImplementsStaticInterface (typeDefintion);
```

Comment:
If we need to cache this, it should be done in `AnnotateMethods` - not in `TryMatchMethod` - ideally `Try` methods should have no side-effects.

Also technically - `TryMatchMethod` is called for both virtual and interface hierarchies. I think we need this new logic only in the interface hierarchy processing.